muikku/src/main/webapp/resources/scripts/gui/environment.js

```diff
@@ -1,6 +1,31 @@
 (function() {
   'use strict';
+
+  /* global converse */
   
+  mApi().chat.status.read().callback(function(err, result) {
+    if (result == null || (result && result.visibility == "VISIBLE_TO_ALL")) {
```

Comment:
Why is `result == null` a passing condition, too? Shouldn't this be `if (result && result.visibility == 'VISIBLE_TO_ALL')` to be sure that chat is only enabled if the status check says so?